### DISCUSSION TITLE
Merging stock tracking history

### DISCUSSION THREAD

**Discussion by adamheinrich** [inventree/InvenTree#5375] [General]
Hello, I am investigating the possibility of using InvenTree for stock tracking at our company. It seems you have managed to create a very nice tool, congratulations!

I understand that InvenTree does not support accounting features but I would like to use it for accounting purposes anyway (to be able to determine the value of stock), either by forking the codebase or using an external tool which would access stock movements via API or plugin and feed back the "correct" prices possibly as parts' internal prices.

So far my plan is something like adding invoice date to the "received against PO" tracking history item which would later enable me to calculate the correct stock value using historical daily exchange rate from the central bank (adding invoice date to PO metadata is not sufficient because a PO can generate multiple shipments/invoices).

However, in the current implementation the tracking history is lost when an item is depleted or when two stock items are merged. While "delete on deplete" can be disabled in settings, merging tracking history is not supported.

So my questions are:

1) Are there any practical considerations which prevent merging of two stock items tracking histories at the moment? Would something break if a merge of two stock items merged their tracking histories as well instead of creating the "MERGED_STOCK_ITEMS" entry for the other item?

2) Would merging of tracking histories be welcome in upstream codebase?

Thank you for your time,

Adam

**Comment by SchrodingersGat** [comment]
Hey @adamheinrich welcome :)

> Would merging of tracking histories be welcome in upstream codebase?

If you can suggest a clean solution to this problem, it would definitely be considered! It's a tricky one.
